checks/normandy/reported_recipes.py

```diff
@@ -0,0 +1,46 @@
+"""
+Recipes available on the server should match the recipes clients are reporting
+Uptake Telemetry about.
+
+The list of recipes for which no event was received is returned. The min/max
+timestamps give the datetime range of the dataset obtained from
+https://sql.telemetry.mozilla.org/queries/64921/
+"""
+import aiohttp
+
+from poucave.typings import CheckResult
+from poucave.utils import fetch_redash
+
+
+EXPOSED_PARAMETERS = ["server"]
+
+REDASH_QUERY_ID = 64921
+
+NORMANDY_URL = "{server}/api/v1/recipe/signed/"
+
+
+async def run(api_key: str, server: str) -> CheckResult:
+    # Fetch latest results from Redash JSON API.
+    rows = await fetch_redash(REDASH_QUERY_ID, api_key)
+
+    min_timestamp = min(r["min_timestamp"] for r in rows)
+    max_timestamp = max(r["max_timestamp"] for r in rows)
+
+    reported_recipes = set(int(r["source"].split("/")[-1]) for r in rows)
+
+    async with aiohttp.ClientSession() as session:
+        # Recipes from source of truth.
+        normandy_url = NORMANDY_URL.format(server=server)
+        async with session.get(normandy_url) as response:
+            normandy_recipes = await response.json()
+
+    normandy_recipes = set(r["recipe"]["id"] for r in normandy_recipes)
+
+    missing = list(normandy_recipes - reported_recipes)
```

Comment:
I think we should also check for `extra`, in case a recipe gets removed but lingers on.

Reply:
The list super long :( 

``` 
{
  "min_timestamp": "2019-09-19T08:59:56.102",
  "max_timestamp": "2019-09-19T14:58:44.180",
  "missing": [],
  "extras": [
    517,
    518,
    530,
    532,
    542,
    576,
    597,
    617,
    636,
    637,
    651,
    662,
    666,
    667,
    668,
    669,
    670,
    674,
    677,
    678,
    680,
    682,
    683,
    692,
    695,
    696,
    698,
    709,
    713,
    714,
    715,
    716,
    717,
    722,
    723,
    726,
    729,
    730,
    731,
    733,
    734,
    736,
    737,
    738,
    739,
    740,
    741,
    742,
    743,
    744,
    745,
    747,
    748,
    749,
    750,
    751,
    752,
    754,
    755,
    756,
    757,
    758,
    759,
    760,
    761,
    763,
    764,
    765,
    768,
    769,
    770,
    772,
    773,
    775,
    776,
    777,
    778,
    779,
    780,
    781,
    782,
    784,
    785,
    787,
    788,
    789,
    790,
    791,
    792,
    793,
    794,
    795,
    796,
    802,
    803,
    804,
    805,
    806,
    807,
    808,
    809,
    810,
    811,
    812,
    813,
    814,
    815,
    816,
    817,
    818,
    819,
    820,
    823,
    826,
    827,
    828,
    829,
    830,
    831,
    833,
    836,
    837,
    841,
    842,
    336,
    372,
    373,
    450,
    451,
    452,
    453,
    454,
    455,
    456,
    483,
    501
  ]
}


```

Reply:
```
common == [259, 783, 786, 531, 661, 797, 799, 800, 801, 672, 553, 686, 687, 688, 689, 690, 691, 821, 824, 825, 832, 834, 835, 838, 839, 840, 457, 843, 844, 845, 718, 719, 720, 460, 721, 746, 753, 634, 767]

extras == [517, 518, 530, 532, 542, 576, 597, 617, 636, 637, 651, 662, 666, 667, 668, 669, 670, 674, 677, 678, 680, 682, 683, 692, 695, 696, 698, 709, 713, 714, 715, 716, 717, 722, 723, 726, 729, 730, 731, 733, 734, 736, 737, 738, 739, 740, 741, 742, 743, 744, 745, 747, 748, 749, 750, 751, 752, 754, 755, 756, 757, 758, 759, 760, 761, 763, 764, 765, 768, 769, 770, 772, 773, 775, 776, 777, 778, 779, 780, 781, 782, 784, 785, 787, 788, 789, 790, 791, 792, 793, 794, 795, 796, 802, 803, 804, 805, 806, 807, 808, 809, 810, 811, 812, 813, 814, 815, 816, 817, 818, 819, 820, 823, 826, 827, 828, 829, 830, 831, 833, 836, 837, 841, 842, 336, 372, 373, 450, 451, 452, 453, 454, 455, 456, 483, 501]